core/jazz_acl/components/validation.js

```diff
@@ -16,23 +16,63 @@
 
 const errorHandlerModule = require("./error-handler.js")();
 
-async function validateInput(userInput) {
-
-		if (!userInput || !userInput.method) {
-			throw (errorHandlerModule.throwInputValidationError("Invalid or missing arguments"));
+function validateBasicInput(userInput) {
+    if (!userInput) {
+      throw (errorHandlerModule.throwInternalServerError("Input parameters are missing"));
+    }
+		if (!userInput.method) {
+			throw (errorHandlerModule.throwInputValidationError("Method is missing"));
     }
 
-    if (!userInput.resourcePath) {
+    if (!userInput.path) {
 			throw (errorHandlerModule.throwInputValidationError("Missing the resource path"));
 		}
 
 		if (!userInput.principalId) {
 			throw (errorHandlerModule.throwForbiddenError("You aren't authorized to access this resource"));
 		}
+}
+
+function validateGetPoliciesInput(userInput) {
+  if (!userInput.query.serviceId) {
+    throw (errorHandlerModule.throwInputValidationError("Service Id is missing"));
+  }
+}
 
-		return true;
+function validatePostPoliciesInput(userInput) {
+  if (!userInput.body) {
+    throw (errorHandlerModule.throwInternalServerError("Body is missing"));
+  }
+
+  if (!userInput.body.serviceId) {
+    throw (errorHandlerModule.throwInputValidationError("Service Id is missing"));
+  }
+
+  if (!userInput.body.policies) {
+    throw (errorHandlerModule.throwInputValidationError("Policy details are missing"));
+  }
+
+  if (userInput.body.policies && userInput.body.policies.length) {
+    let missingFields = [];
+    userInput.body.policies.forEach(policy => {
+      if (!policy.userId) {
+        missingFields.push("userId");
+      }
+      if (!policy.permission) {
+        missingFields.push("permission");
+      }
+      if (!policy.category) {
+        missingFields.push("category");
+      }
+    });
+    if (missingFields.length) {
+      throw (errorHandlerModule.throwInputValidationError(`Policy details are missing values for ${missingFields.join(', ')}`));
```

Comment:
Since you are doing this for an array of policies, this message can get messy making it difficult the understand the real error (Ex: `Policy details are missing values for userId, userId, userId, permission, category, category`)

Let's keep it simple by saying these fields are required in each policy.